<a href="https://colab.research.google.com/github/luizfernando120/aaaaaaaaaaaa/blob/main/Projeto_07_Abril.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Bibliotecas


In [47]:
import numpy as np
import pandas as pd
import plotly.express as px
import matplotlib.pyplot as plt


## **Numpy**
### numpy é uma biblioteca Python que fornece um objeto de matriz multidimensional,vários objetos derivados (como matrizes e matrizes mascaradas) e uma variedadede rotinas para operações rápidas em matrizes. (**CAPARROZ**, 2023)
## **Pandas**
### pandas é uma biblioteca de software livre para a linguagem de programação,Python, utilizada principalmente para manipulação e análise de dados;Ele fornece uma estrutura de dados flexível e de alto desempenho, chamada DataFrame, que é capaz de lidar com grandes conjuntos de dados e realizar operações comuns de análise de dados, como limpeza, filtragem, agrupamento, agregação, transformação e visualização. (**CAPARROZ**, 2023)
## **Plotly Express**
### plotly é uma biblioteca de visualização de dados para Python, Javascript e R. Eles tem uma série de produtos, desde para criação de dashboards até clientes SQL. Plotly permite que você utilize seus gráficos em aplicações e, claro, Jupyter Notebooks. (**VASCONCELLOS**, 2018)


#Importação dos dados

In [4]:
df = pd.read_csv ('https://raw.githubusercontent.com/luizfernando120/Projeto_Governanca_TI_2023/main/obitos_publico.csv', sep = ';')

# Dados referentes aos acidentes fatais de trânsito no estado de São Paulo

In [5]:
df

Ano do BO (RDO) Data do Acidente Tipo do veiculo da vitima  \
0                 2023       28/02/2023               MOTOCICLETA   
1                 2023       28/02/2023               MOTOCICLETA   
2                 2023       28/02/2023                  PEDESTRE   
3                 2023       28/02/2023            NAO DISPONIVEL   
4                 2023       05/02/2023                  CAMINHAO   
...                ...              ...                       ...   
42826             2014       27/07/2014                 AUTOMOVEL   
42827             2015       01/01/2015                 BICICLETA   
42828             2015       01/01/2015                 BICICLETA   
42829             2015       01/01/2015                  PEDESTRE   
42830             2015       01/01/2015               MOTOCICLETA   

       Tipo de vitima Tipo de acidente       Sexo  Idade da vitima  
0            CONDUTOR   NAO DISPONIVEL  MASCULINO               21  
1            CONDUTOR          COLISAO  MASCULINO               55  
2            PEDESTRE    ATROPELAMENTO   FEMININO               89  
3      NAO DISPONIVEL    ATROPELAMENTO   FEMININO               78  
4      NAO DISPONIVEL          COLISAO  MASCULINO               35  
...               ...              ...        ...              ...  
42826      PASSAGEIRO          COLISAO   FEMININO               42  
42827        CONDUTOR          COLISAO  MASCULINO               42  
42828        CONDUTOR           CHOQUE  MASCULINO               44  
42829        PEDESTRE    ATROPELAMENTO  MASCULINO               66  
42830        CONDUTOR           OUTROS  MASCULINO               27  

[42831 rows x 7 columns]

#Ocorrências agrupadas por sexo

In [6]:
quantidade_tipos_sexo = df['Sexo'].value_counts()

In [7]:
quantidade_tipos_sexo

MASCULINO         34801
FEMININO           7563
NAO DISPONIVEL      467
Name: Sexo, dtype: int64

In [8]:
quantidade_tipos_sexo.index

Index(['MASCULINO', 'FEMININO', 'NAO DISPONIVEL'], dtype='object')

In [9]:
quantidade_tipos_sexo.values

array([34801,  7563,   467])

In [10]:
sexos = ['MASCULINO','FEMININO','NAO DISPONIVEL']
count_sexos = [34801,7563,467]

In [58]:
fig = px.bar(x=sexos,y=count_sexos,orientation='v',color=sexos,width=600,height=600)
fig

Com base nas informações fornecidas, podemos concluir preliminarmente que os acidentes de trânsito são mais frequentes entre os indivíduos do sexo masculino sendo **34.801**, enquanto o gênero feminino apresenta uma proporção menor de **7.563** ocorrências. É importante ressaltar que essa análise é baseada apenas nos dados fornecidos e não leva em consideração outros fatores que podem influenciar os acidentes de trânsito.
Já não se verificou o número de **467** dos acidentes fatais, por ausência de informações, privacidade, falta de perguntas.


In [12]:
px.pie(quantidade_tipos_sexo, names=quantidade_tipos_sexo.index, values=quantidade_tipos_sexo.values,width=900,height=600)

Com base no gráfico acima, podemos observar que o sexo *Masculino* está mais presente dentre os acidentes fatais, representando cerca de *81,3%* dos acidentes enquanto o sexo *Feminino* representa cerca de *17,7%*.

**Esta figura é apenas uma forma diferente de ilustrar os dados fornecidos no gráfico quantitativo dessas informações.**

#MEDIA

In [13]:
df_agrupado_pelo_sexo_media = df.groupby('Sexo').agg({ 'Idade da vitima': np.mean }).reset_index()

In [14]:
df_agrupado_pelo_sexo_media

Sexo  Idade da vitima
0        FEMININO        43.937062
1       MASCULINO        40.742795
2  NAO DISPONIVEL        40.278373

 Já no quesito, Idade podemos observar no gráfico que dentre as ocorrências, a **média** de idade entre as mulheres (43,9 anos) é a maior nos acidentes fatais no estado de São Paulo.

#MEDIANA

In [15]:
df_agrupado_pelo_sexo_mediana = df.groupby('Sexo').agg({ 'Idade da vitima': np.median }).reset_index()

In [16]:
df_agrupado_pelo_sexo_mediana

Sexo  Idade da vitima
0        FEMININO             41.0
1       MASCULINO             38.0
2  NAO DISPONIVEL             35.0

 Quando calculamos a **mediana** de idades, podemos observa que o resultado é semelhante a média, sendo as mulheres (41 anos) com mais idade nos acidentes fatais no estado de São Paulo.

#MODA

In [34]:
data = pd.Series (df['Idade da vitima'])
moda = data.mode()
moda

0    21
Name: Idade da vitima, dtype: int64

Já quando nos atentamos a *moda* podemos observar que a idade com maior ocorrência de acidentes fatais nas estradas está na casa dos *21 anos*, considerada por muitos como uma idade de risco.


#DESVIO PADRAO

In [39]:
desvio_padrao =np.std(data)
round(desvio_padrao, 2)

18.52

Podemos observar que o desvio padrão dos dados é de *18.52 anos*.

**Esta medida representa a dispersão dos dados em relação à *média*.**


#Ocorrências agrupadas por Tipo de Veiculo



In [19]:
quantidade_tipos_veiculos = df['Tipo do veiculo da vitima'].value_counts()

In [20]:
quantidade_tipos_veiculos

MOTOCICLETA       15523
AUTOMOVEL         10827
PEDESTRE           9865
BICICLETA          2807
NAO DISPONIVEL     1789
CAMINHAO           1444
ONIBUS              353
OUTROS              223
Name: Tipo do veiculo da vitima, dtype: int64

In [21]:
quantidade_tipos_veiculos.index

Index(['MOTOCICLETA', 'AUTOMOVEL', 'PEDESTRE', 'BICICLETA', 'NAO DISPONIVEL',
       'CAMINHAO', 'ONIBUS', 'OUTROS'],
      dtype='object')

In [22]:
quantidade_tipos_veiculos.values

array([15523, 10827,  9865,  2807,  1789,  1444,   353,   223])

In [23]:
veiculos =['MOTOCICLETA', 'AUTOMOVEL', 'PEDESTRE', 'BICICLETA', 'NAO DISPONIVEL', 'CAMINHAO', 'ONIBUS', 'OUTROS']
quantidade = [15523, 10827,  9865,  2807,  1789,  1444,   353,   223]

In [40]:
px.bar(x=veiculos,y=quantidade,orientation='v',color=veiculos,width=1500,height=600)

Esses dados fornecem insights valiosos para entender a distribuição dos acidentes fatais por tipo de veículo envolvido.

Observamos que os acidentes fatais envolvendo motocicletas têm a maior frequência, representando o tipo de veículo com o maior número de ocorrências fatais. Isso sugere que os acidentes de motocicleta, sendo 15523, uma preocupação significativa em termos de segurança no trânsito. É necessário um foco especial em estratégias de prevenção voltadas para os usuários de motocicletas, como campanhas de conscientização, treinamento adequado e fiscalização.

Os acidentes fatais envolvendo automóveis (10827) e pedestres (9865) também possuem números expressivos. Esses dados destacam a importância de medidas de segurança para motoristas e a necessidade de melhorias na infraestrutura urbana para proteger os pedestres.

Já os acidentes fatais envolvendo bicicletas, caminhões, ônibus e outros veículos têm uma incidência menor, mas ainda assim não devem ser negligenciados.


In [25]:
px.pie(quantidade_tipos_veiculos, names=quantidade_tipos_veiculos.index, values=quantidade_tipos_veiculos.values,width=1200,height=600)

**Distribuição Percentual dos Veículos Envolvidos:**

**Motocicleta:** Representando 36,2% das ocorrências, as motocicletas são responsáveis pela maioria dos acidentes fatais.

**Automóvel:** Com 25,3% das ocorrências, os automóveis também têm uma parcela significativa nos acidentes fatais.

**Pedestre:** Correspondendo a 23% das ocorrências, os pedestres são o terceiro grupo mais afetado.

**Bicicleta:** Com 6,55% das ocorrências, as bicicletas também contribuem para os acidentes fatais.

**Não disponível:** Com 4,18% das ocorrências sem especificação do tipo de veículo, a falta de informação dificulta a análise precisa.

**Caminhão:** Representando 3,37% das ocorrências, os acidentes envolvendo caminhões são menos frequentes.

**Ônibus:** Com 0,824% das ocorrências, os ônibus têm uma porcentagem relativamente baixa de envolvimento em acidentes fatais.

**Outros:** Correspondendo a 0,521% das ocorrências, outros tipos de veículos apresentam uma participação mínima.


#Ocorrências agrupadas por ano


In [26]:
quantidade_anos = df['Ano do BO (RDO)'].value_counts()

In [27]:
quantidade_anos

2015    6091
2016    5674
2017    5410
2022    5296
2019    5205
2018    5179
2020    4678
2021    4554
2023     704
2014      26
2013       8
2012       4
2011       1
2007       1
Name: Ano do BO (RDO), dtype: int64

In [28]:
quantidade_anos.index

Int64Index([2015, 2016, 2017, 2022, 2019, 2018, 2020, 2021, 2023, 2014, 2013,
            2012, 2011, 2007],
           dtype='int64')

In [29]:
quantidade_anos.values

array([6091, 5674, 5410, 5296, 5205, 5179, 4678, 4554,  704,   26,    8,
          4,    1,    1])

In [30]:
anos =['2007', '2011', '2012', '2013',  '2014', '2015', '2016', '2017', '2018', '2019', '2020','2021' , '2022' , '2023' ]
numero = [ 1, 1, 4, 8, 26, 6091,  5674, 5410, 5179, 5205 ,4678 , 4554,  5296, 704]

In [31]:
fig = px.scatter(x=anos,y=numero)
fig.update_traces(mode='lines+markers')


O ano com o maior número de ocorrências registradas é 2015, com 6.091 acidentes fatais.

Os anos subsequentes, 2016 e 2017, também apresentam uma quantidade significativa de ocorrências, com 5.674 e 5.410 acidentes fatais, respectivamente.

Os anos mais recentes, 2022 e 2019, ainda registram uma quantidade considerável de acidentes fatais, com 5.296 e 5.205 ocorrências, respectivamente.

Por outro lado, os anos anteriores a 2015 apresentam números muito baixos de acidentes fatais, indicando uma possível ausência de monitoramento e registro dos acidentes fatais.

**2023 apresentam poucos acidentes, pois os dados analisados, estão referentes apenas ate o mês de Fevereiro**

Essa análise agrupada por ano permite observar as tendências de ocorrências ao longo do tempo.


